sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -2057,14 +2055,17 @@ case class ValidateExternalType(child: Expression, expected: DataType, externalD
     if (checkType(input)) {
       input
     } else {
-      throw new RuntimeException(s"${input.getClass.getName}$errMsg")
+      throw new SparkRuntimeException("INVALID_EXTERNAL_TYPE",
+        Map(
+          "actualType" -> input.getClass.getName,
+          "expectedType" -> expected.simpleString,
+          "childExpression" -> child.toString
```

Comment:
```suggestion
          "childExpression" -> toSQLExpr(child)
```

Reply:
Thank you. 

Applyied at `QueryExecutionErrors`:

```
  def invalidExternalTypeError(
      actualType: String, expectedType: String, childExpression: Expression): SparkRuntimeException = {
    new SparkRuntimeException(
      errorClass = "INVALID_EXTERNAL_TYPE",
      messageParameters = Map(
        "actualType" -> actualType,
        "expectedType" -> expectedType,
        "childExpression" -> toSQLExpr(childExpression)
      )
    )
  }
```